services/app-web/src/pages/StateSubmission/Documents/Documents.test.tsx

```diff
@@ -357,6 +385,85 @@ describe('Documents', () => {
         })
     })
 
+    describe('the delete button', () => {
+        // For this test, we want to mock the call to deleteFile to see when it gets called
+        const mockS3 = testS3Client()
+        const deleteCallKeys: string[] = []
+        mockS3.deleteFile = async (key) => {
+            console.log('MOCK DELETE CALLED', key)
+            deleteCallKeys.push(key)
+        }
+
+        it('does not delete files from past revisions', async () => {
+            const submission = mockUnlockedHealthPlanPackageWithDocuments()
+            renderWithProviders(
+                <Route
+                    path={RoutesRecord.SUBMISSIONS_FORM}
+                    component={StateSubmissionForm}
+                />,
+                {
+                    apolloProvider: {
+                        mocks: [
+                            fetchCurrentUserMock({ statusCode: 200 }),
+                            fetchHealthPlanPackageMock({
+                                id: '15',
+                                statusCode: 200,
+                                submission,
+                            }),
+                        ],
+                    },
+                    routerProvider: { route: '/submissions/15/documents' },
+                    s3Provider: mockS3,
+                }
+            )
+
+            // PERFORM
+
+            // we should be able to find delete buttons for each of the three recent files.
+            // the aria label for each button is a lifesaver here.
+            const removeOneTwo = await screen.findByLabelText(
+                'Remove one two document'
+            )
+            userEvent.click(removeOneTwo)
```

Comment:
I’d add a couple more clicks here to make sure that it works across different revisions. Part of the reason I built the mock to write to an array is you could delete all the files and then assert that only the ones you expected got deleted.